internal/pkg/repository/repository.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package repository provides support for building and pushing images to a repository.
+package repository
+
+import (
+    "fmt"
+)
+
+// ContainerLoginBuildPusher provides support for logging in to repositories, building images and pushing images to repositories.
+type ContainerLoginBuildPusher interface {
+    Build(uri, path, imageTag string, additionalTags ...string) error
+    Login(uri, username, password string) error
+    Push(uri, imageTag string, additionalTags ...string) error
+}
+
+// Registry gets information of repositories.
+type Registry interface {
+    RepositoryURI(name string) (string, error)
+    Auth() (string, string, error)
```

Comment:
can we make it 
```suggestion
    Auth() (username string, password string, err error)
```
to make the input clearer?